front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts

```diff
@@ -94,26 +100,28 @@ async function handler(
       }
 
       const { name, description, csv } = bodyValidation.right;
-      const csvRowsRes = await rowsFromCsv(csv);
-      if (csvRowsRes.isErr()) {
+      const csvRowsRes = csv ? await rowsFromCsv(csv) : null;
+
+      if (csvRowsRes?.isErr()) {
         return apiError(req, res, {
           api_error: csvRowsRes.error,
           status_code: 400,
         });
       }
 
-      const csvRows = csvRowsRes.value;
-      if (csvRows.length > 500_000) {
+      const csvRows = csvRowsRes?.unwrap();
```

Comment:
needed the unwrap feature as checking `csvRowsRes?.isErr()` above doesn't do the type narrowing as one would hope :/

Reply:
Yes this would work 👍